From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martijn Muijsers <martijnmuijsers@live.nl>
Date: Mon, 10 Jul 2023 22:05:08 +0200
Subject: [PATCH] Modifiable Bukkit enums - Remove hard-coded properties -
 Material - Id

Adds a test for equality between `Material.getId` and its original backing implementation.

License: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)
Fiddle - https://fiddlemc.org

diff --git a/src/test/java/org/fiddlemc/fiddle/material/legacy/PerLegacyMaterialTest.java b/src/test/java/org/fiddlemc/fiddle/material/legacy/PerLegacyMaterialTest.java
index b5162d77e33a806a74b24233ad38127029071702..903a76b0ca91d43269282939d79a92851ce86961 100644
--- a/src/test/java/org/fiddlemc/fiddle/material/legacy/PerLegacyMaterialTest.java
+++ b/src/test/java/org/fiddlemc/fiddle/material/legacy/PerLegacyMaterialTest.java
@@ -37,6 +37,16 @@ public class PerLegacyMaterialTest extends AbstractTestingBase {
         }
     }
 
+    // Fiddle start - modifiable Bukkit enums - Material - remove hard-coded properties - id
+    @Test
+    @SuppressWarnings("deprecation")
+    public void getId() {
+        for (Material material : getMaterials()) {
+            assertEquals(material.getIdLegacy(), material.getId(), "getId does not equal legacy getId for " + material);
+        }
+    }
+    // Fiddle end - modifiable Bukkit enums - Material - remove hard-coded properties - id
+
     // Fiddle start - modifiable Bukkit enums - Material - remove hard-coded properties - max stack size
     @Test
     public void getMaxStackSize() {
